tests/mtouch/RegistrarTest.cs

```diff
@@ -1843,6 +1843,32 @@ public SomeNativeObject M2 ()
 				mtouch.AssertWarning (4179, $"The registrar found the abstract type 'SomeNativeObject' in the signature for 'C.M2'. Abstract types should not be used in the signature for a member exported to Objective-C.", "testApp.cs", 21);
 			}
 		}
+
+		// https://github.com/xamarin/xamarin-macios/issues/7733
+		[Test]
+		public void GHIssue7733 ()
+		{
+			var str1 = @"
+public delegate void ACompletionHandler (string strArg, NSError error);
+
+[Preserve]
+public class WidgetObject : NSObject {
+	[Export (""doSomeWork:completion:"")]
+	public virtual void DoWork (string who, ACompletionHandler completion)
```

Comment:
I've usually put positive registrar tests (code that's supposed to work, not code that's supposed to show an error) in tests/monotouch-test/ObjCRuntime/RegistrarTest, because that's much faster to execute (this test adds a whole project build to the tests, while in tests/monotouch-test/ObjCRuntime/RegistrarTest the additional work is just to compile a few more lines of C#).

Reply:
Thanks, will move them 👍